python/src/tiff.rs

```diff
@@ -69,7 +69,11 @@ impl PyTIFF {
             // TODO: avoid this clone; add Arc to underlying rust code?
             .clone();
         future_into_py(py, async move {
-            let tile = ifd.fetch_tile(x, y, reader.as_ref()).await.unwrap();
+            let tile = ifd
+                .fetch_tile(x, y, reader.as_ref())
+                .await
+                .map_err(|err| PyTypeError::new_err(err.to_string()))?;
+
```

Comment:
Ideally we'd have an error enum with known variants so that we can check specifically for "trying to load a strip tiff as a tiled tiff" and return a Python TypeError for that specific case.

But this is fine for now.